### PR TITLE
[TR] PIM-4781: Move User class to PIM

### PR DESCRIPTION
**Pull request by solivier**
Move the user class into the pim user bundle instead of the oro user bundle in order to modify the user class
- [ ] rebase commits

### CODE REVIEW

File: src/Oro/Bundle/ConfigBundle/Tests/Unit/Config/UserConfigManagerTest.php [hunk 1]
```diff
@@ -6,7 +6,7 @@
 use Doctrine\Common\Persistence\ObjectRepository;
 
 use Oro\Bundle\ConfigBundle\Config\UserConfigManager;
-use Oro\Bundle\UserBundle\Entity\User;
+use Pim\Bundle\UserBundle\Entity\User;
```
Comment (juliensnz) [inline src/Oro/Bundle/ConfigBundle/Tests/Unit/Config/UserConfigManagerTest.php:9]: we cannot reference the pim in the platform :/

File: src/Oro/Bundle/NavigationBundle/Controller/Api/NavigationItemController.php [hunk 2]
```diff
@@ -10,7 +10,7 @@
 use FOS\Rest\Util\Codes;
 use Nelmio\ApiDocBundle\Annotation\ApiDoc;
 
-use Oro\Bundle\UserBundle\Entity\User;
+use Pim\Bundle\UserBundle\Entity\User;
```
Comment (juliensnz) [inline src/Oro/Bundle/NavigationBundle/Controller/Api/NavigationItemController.php]: same here

File: src/Oro/Bundle/ConfigBundle/Tests/Unit/Config/UserConfigManagerTest.php (hunk 1 above)
Comment (nidup) [inline src/Oro/Bundle/ConfigBundle/Tests/Unit/Config/UserConfigManagerTest.php:9]: we do as we prepare the move

**Comment by grena** [comment]
GTM

**Comment by phaseinducer** [comment]
GTM

**Pull request merged**